models/stocks.js

```diff
@@ -38,7 +40,80 @@ const fetchStocks = async () => {
   }
 }
 
+/**
+ * Fetches the user stocks
+ * @return {Promise<userStocks|object>}
+ */
+const fetchUserStocks = async (userId, stockId = null) => {
+  try {
+    let userStocksRef = ''
+    const query = userStocksModel.where('userId', '==', userId)
+    if (stockId) {
+      userStocksRef = await query.where('stockId', '==', stockId).get()
+      const [userStocks] = userStocksRef.docs
+      if (userStocks) {
+        return { id: userStocks.id, ...userStocks.data() }
+      }
+      return {}
+    }
+
+    userStocksRef = await query.get()
+    const userStocks = []
+    userStocksRef.forEach((stock) => {
+      userStocks.push({
+        id: stock.id,
+        ...stock.data()
+      })
+    })
+    return userStocks
+  } catch (err) {
+    logger.error('Error retrieving user stocks', err)
+    throw err
+  }
+}
+
+/**
+ * Create user stocks
+ * @return {Promise<userStocks|object>}
+ */
+const createUserStock = async (userId, stockData) => {
+  try {
+    const userStocks = {
+      userId,
+      ...stockData
+    }
+    const { id } = await userStocksModel.add(userStocks)
+    return { id }
+  } catch (err) {
+    logger.error('Error creating user stocks', err)
+    throw err
+  }
+}
+
+/**
+ * Update Users Stocks
+ * @return {Promise<userStocks|object>}
+ */
+const updateUserStocks = async (userId, stockData) => {
```

Comment:
We can simply create an `addOrUpdate` function here and remove the above function, as this function is also handling the adding stocks part.

Reply:
This function handles only updating of stocks. I created the above `createUserStocks` function just to make the code look cleaner and readable.

Reply:
4668438